tests/integration/backup_tests/test_backups.py

```diff
@@ -231,6 +241,143 @@ async def test_multi_backup(ops_test: OpsTest, add_writes_to_db) -> None:
+    most_recent_backup = list_result.split("\n")[-1]
+    backup_id = most_recent_backup.split()[0]
+    action = await db_unit.run_action(action_name="restore", **{"backup-id": backup_id})
+    restore = await action.wait()
+    assert restore.results["restore-status"] == "restore started", "restore not successful"
+
+    async with ops_test.fast_forward():
+        await asyncio.gather(
+            ops_test.model.wait_for_idle(apps=[db_app_name], status="active"),
+        )
+
+    # verify all writes are present
+    try:
+        for attempt in Retrying(stop=stop_after_delay(4), wait=wait_fixed(20)):
+            with attempt:
+                number_writes_restored = await ha_helpers.count_writes(ops_test)
+                assert number_writes == number_writes_restored, "writes not correctly restored"
+    except RetryError:
+        assert number_writes == number_writes_restored, "writes not correctly restored"
+
+
+@pytest.mark.parametrize("cloud_provider", ["AWS", "GCP"])
+async def test_restore_new_cluster(ops_test: OpsTest, add_writes_to_db, cloud_provider):
+    # configure test for the cloud provider
+    db_app_name = await helpers.app_name(ops_test)
+    await helpers.set_credentials(ops_test, cloud=cloud_provider)
+    if cloud_provider == "AWS":
+        configuration_parameters = {
+            "bucket": "pbm-test-bucket-1",
+            "region": "us-west-2",
+            "endpoint": "",
+        }
+    else:
+        configuration_parameters = {
+            "bucket": "data-charms-testing",
+            "endpoint": "https://storage.googleapis.com",
+            "region": "",
+        }
+
+    await ops_test.model.applications[S3_APP_NAME].set_config(configuration_parameters)
+    await asyncio.gather(
+        ops_test.model.wait_for_idle(apps=[S3_APP_NAME], status="active"),
+        ops_test.model.wait_for_idle(apps=[db_app_name], status="active"),
+    )
+
+    # create a backup
+    writes_in_old_cluster = await ha_helpers.count_writes(ops_test, db_app_name)
+    assert writes_in_old_cluster > 0, "old cluster has no writes."
+    await helpers.create_and_verify_backup(ops_test)
+
+    # save old password, since after restoring we will need this password to authenticate.
+    old_password = await ha_helpers.get_password(ops_test, db_app_name)
+
+    # remove the old cluster and make a new cluster with the same name. Backups with PBM are
+    # incompatible if their cluster names differ.
+    await helpers.destory_cluster(ops_test, db_app_name)
+
+    db_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(db_charm, num_units=3, application_name=db_app_name)
+    await asyncio.gather(
+        ops_test.model.wait_for_idle(apps=[db_app_name], status="active"),
+    )
+
+    db_unit = await helpers.get_leader_unit(ops_test)
+    action = await db_unit.run_action("set-password", **{"password": old_password})
+    action = await action.wait()
+    assert action.status == "completed"
```

Comment:
I don't think it is a necessary step for MongoDB. Can we avoid it?

Reply:
I would need to think of how do avoid this. Perhaps there is a way to make a back up but exclude specific things in the admin DB like `password`. 

Right now the restore will restore things like password and replica set name, which means we are required to use the same replica set name and password

Reply:
As discussed via MM, we decided this was "okay-ish"